docs(release): add 1.7.0 release note (#9259)
longhorn/longhorn: CHANGELOG/CHANGELOG-1.7.0.md

## Longhorn v1.7.0 Release Notes

This latest version of Longhorn introduces several features, enhancements, and bug fixes that are intended to improve system quality and the overall user experience. Highlights include new V2 Data Engine features, platform-agnostic deployment, high availability, and improvements to data protection, stability, performance, and resilience.

The Longhorn team appreciates your contributions and anticipates receiving feedback regarding this release.

> **Note:**
> For more information about release-related terminology, see [Releases](https://github.com/longhorn/longhorn#releases).

## Primary Highlights

### New V2 Data Engine Features

> Although the V2 Data Engine is still considered a preview feature in this release, the core functions have been significantly enhanced. For example, enhancements include online replica rebuilding, filesystem trim feature, which allows the system to reclaim unused space more effectively. Block-type bisk, enabling the use of SPDK AIO, NVMe, and VirtIO Bdev drivers, which broadens the range of compatible hardware and improves performance in high-demand environments.

Reviewer: `Block-type disk` instead of `Block-type bisk`

Author: Fixed. Thank you.

## Longhorn v1.7.0 Release Notes

This latest version of Longhorn introduces several features, enhancements, and bug fixes that are intended to improve system quality and the overall user experience. Highlights include new V2 Data Engine features, platform-agnostic deployment, high availability, and improvements to data protection, stability, performance, and resilience.

The Longhorn team appreciates your contributions and anticipates receiving feedback regarding this release.

> **Note:**
> For more information about release-related terminology, see [Releases](https://github.com/longhorn/longhorn#releases).

## Deprecation

The functionality of the [environment check script](https://github.com/longhorn/longhorn/blob/v1.7.x/scripts/environment_check.sh) overlaps with that of the Longhorn CLI, which is available starting with v1.7.0. Because of this, the script is deprecated in v1.7.0 and is scheduled for removal in v1.8.0.

## Primary Highlights

### New V2 Data Engine Features

> Although the V2 Data Engine is still considered a preview feature in this release, the core functions have been significantly enhanced. For example, enhancements include online replica rebuilding, filesystem trim feature, which allows the system to reclaim unused space more effectively. Block-type disk, enabling the use of SPDK AIO, NVMe, and VirtIO Bdev drivers, which broadens the range of compatible hardware and improves performance in high-demand environments.

Reviewer 2: Suggest mentioning the ongoing efforts on volume live upgrade, even though it's just half way. In  v1.7.0, we primarily focus on the implementation on data plane and will continue the control plane part to complete the feature in following releases.

Author: Added. Thank you.

## Longhorn v1.7.0 Release Notes

This latest version of Longhorn introduces several features, enhancements, and bug fixes that are intended to improve system quality and the overall user experience. Highlights include new V2 Data Engine features, platform-agnostic deployment, high availability, and improvements to data protection, stability, performance, and resilience.

The Longhorn team appreciates your contributions and anticipates receiving feedback regarding this release.

> **Note:**
> For more information about release-related terminology, see [Releases](https://github.com/longhorn/longhorn#releases).

## Deprecation

The functionality of the [environment check script](https://github.com/longhorn/longhorn/blob/v1.7.x/scripts/environment_check.sh) overlaps with that of the Longhorn CLI, which is available starting with v1.7.0. Because of this, the script is deprecated in v1.7.0 and is scheduled for removal in v1.8.0.

## Primary Highlights

### New V2 Data Engine Features

> Although the V2 Data Engine is still considered a preview feature in this release, the core functions have been significantly enhanced. For example, enhancements include online replica rebuilding, filesystem trim feature, which allows the system to reclaim unused space more effectively. Block-type disk, enabling the use of SPDK AIO, NVMe, and VirtIO Bdev drivers, which broadens the range of compatible hardware and improves performance in high-demand environments. Regarding live upgrades, Longhorn 1.7.0 focuses primarily on the data plane, with plans to will continue the control plane part in future releases to fully complete the feature.
> 
> - [Online Replica Rebuilding](https://github.com/longhorn/longhorn/issues/7199)
> - [Filesystem Trim](https://github.com/longhorn/longhorn/issues/7534)
> - [Block-type Disk Supports SPDK AIO, NVMe and VirtIO Bdev Drivers](https://github.com/longhorn/longhorn/issues/7672)
> - [V2 Volume Supports Live Upgrade for Data Plane](https://github.com/longhorn/longhorn/issues/6001)

Reviewer 3: ```suggestion
Although the V2 Data Engine is still considered a preview feature in this release, the core functions have been significantly enhanced.

- [Online replica rebuilding](https://github.com/longhorn/longhorn/issues/7199): Allows Longhorn to rebuild replicas while the volume is running without any IO interruptions.
- [Filesystem trim](https://github.com/longhorn/longhorn/issues/7534): Reclaims unused space more effectively.
- [Block-type disk support for SPDK AIO, NVMe and VirtIO Bdev drivers](https://github.com/longhorn/longhorn/issues/7672): Broadens the range of compatible hardware and improves performance in high-demand environments.
- [V2 volume support for data plane live upgrade](https://github.com/longhorn/longhorn/issues/6001): Enables upgrading of the spdk_tgt process, which handles IO operations, without any downtime. Coverage of the live upgrade function will be extended to the control plane in a future release.
```

The Longhorn team will continue to develop features for the V1 Data Engine and to prepare the V2 Data Engine for use in all types of environments.

### Platform-Agnostic Deployment

Longhorn is designed to seamlessly operate on general-purpose Linux distributions, and on certain container-optimized systems. Starting with Longhorn v1.7.0, Longhorn supports container-optimized OS (COS), providing robust and efficient persistent storage solutions for Kubernetes clusters running on COS.

- [Container-Optimized OS support](https://github.com/longhorn/longhorn/issues/6165) 

### High Availability

> Starting with v1.7.0, Longhorn introduced key features to enhance system resilience and address potential single points of failure. High availability for backing images is implemented to mitigate the risks associated with their failure. The RWX Volumes fast failover feature is added to improve resilience against share-manager pod failures, allowing for rapid detection and response independent of the timing and sequence of Kubernetes node failures.
> 
> - [HA Backing Image](github.com/longhorn/longhorn/issues/2856)
> - [RWX Volumes Fast Failover](https://github.com/longhorn/longhorn/issues/6205)

Reviewer 3: ```suggestion
v1.7.0 introduces features that enhance system resilience and address potential single points of failure. 

- [High availability for backing images](github.com/longhorn/longhorn/issues/2856): Mitigates risks associated with image failures. 
- [RWX volumes fast failover](https://github.com/longhorn/longhorn/issues/6205): Enables rapid detection of and response to Share Manager pod failures (independent of the timing and sequence of Kubernetes node failures).
```

### Data Protection

> Since v1.7.0, periodic and on-demand full backups have been supported to enhance backup reliability. It decreases the likelihood of backup corruption and enhances the overall reliability of the backup process.
> 
> - [Support Periodic or On-Demand Full Backups To Enhance Backup Reliability](https://github.com/longhorn/longhorn/issues/7070)

Reviewer 3: ```suggestion
Starting with v1.7.0, Longhorn supports [periodic and on-demand full backups](https://github.com/longhorn/longhorn/issues/7070) that help reduce the likelihood of backup corruption and enhance the overall reliability of the backup process.
```

### Scheduling

> The replica auto-balancing feature is enhanced to address disk space pressure from growing volumes. These enhancements reduce manual intervention by automatically rebalancing replicas under disk pressure and improve performance with faster replica rebuilding through local file copying.
> 
> - [Auto-Balancing Volumes Between Nodes and Disks](https://github.com/longhorn/longhorn/issues/4105)

Reviewer 3: ```suggestion
The [replica auto-balancing](https://github.com/longhorn/longhorn/issues/4105) feature was enhanced to address disk space pressure from growing volumes. These enhancements reduce manual intervention by automatically rebalancing replicas under disk pressure and improve performance with faster replica rebuilding through local file copying.
```

### Storage Network

> The storage network now supports RWX volumes with network segregation, enabling dedicated traffic lanes for storage operations.
> 
> - [Support storage network for RWX volumes](https://github.com/longhorn/longhorn/issues/8184)

Reviewer 3: ```suggestion
The storage network now [supports RWX volumes]((https://github.com/longhorn/longhorn/issues/8184)) with network segregation, enabling dedicated traffic lanes for storage operations.
```

> ### Command Line Tool
> 
> The Longhorn CLI which is the official Longhorn command line tool, is introduced in v1.7.0. This tool interacts with Longhorn by creating Kubernetes custom resources (CRs) and executing commands inside a dedicated pod for in-cluster and host operations. Usage scenarios include installation, operations such as exporting replicas, and troubleshooting.
> 
> - [Create Longhorn CLI Focusing on Non CR Resource Operations](https://github.com/longhorn/longhorn/issues/7927)

Reviewer 3: ```suggestion
### Longhorn CLI

The [Longhorn CLI](https://github.com/longhorn/longhorn/issues/7927), which is the official Longhorn command line tool, is introduced in v1.7.0. This tool interacts with Longhorn by creating Kubernetes custom resources (CRs) and executing commands inside a dedicated pod for in-cluster and host operations. Usage scenarios include installation, operations such as exporting replicas, and troubleshooting.
```
